### ISSUE TITLE
Newbie question:   java.io.IOException: Invalid argument 

### ISSUE THREAD

**Issue by bwalsh** [SciCrunch/SciGraph#97]
Hi.
Just started here at OHSU.  I'm attempting to install and launch this locally.

Getting the following error when executing the "Build the graph" step

```
mvn exec:java -Dexec.mainClass="edu.sdsc.scigraph.owlapi.loader.BatchOwlLoader" -Dexec.args="-c src/test/resources/pizzaExample.yaml"
```

System info ( I'm running vagrant & virtualbox )

```
$ service neo4j-service status
 * neo4j is running
$ java -version
java version "1.7.0_67"
Java(TM) SE Runtime Environment (build 1.7.0_67-b01)
Java HotSpot(TM) 64-Bit Server VM (build 24.65-b04, mixed mode)

$ more /etc/*-release
::::::::::::::
/etc/lsb-release
::::::::::::::
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=12.04
DISTRIB_CODENAME=precise
DISTRIB_DESCRIPTION="Ubuntu 12.04.2 LTS"
```

The error:

Apr 03, 2015 11:08:25 PM edu.sdsc.scigraph.owlapi.loader.OwlLoaderModule getInserter
INFO: Getting BatchInserter
[WARNING]
java.lang.reflect.InvocationTargetException
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:606)
        at org.codehaus.mojo.exec.ExecJavaMojo$1.run(ExecJavaMojo.java:293)
        at java.lang.Thread.run(Thread.java:745)
Caused by: java.io.IOError: java.io.IOException: Invalid argument
        at org.mapdb.Volume$MappedFileVol.makeNewBuffer(Volume.java:499)
        at org.mapdb.Volume$MappedFileVol.<init>(Volume.java:433)
        at org.mapdb.Volume.volumeForFile(Volume.java:176)
        at org.mapdb.Volume$1.createIndexVolume(Volume.java:203)
        at org.mapdb.StoreDirect.<init>(StoreDirect.java:202)
        at org.mapdb.DBMaker.extendStoreDirect(DBMaker.java:918)
        at org.mapdb.DBMaker.makeEngine(DBMaker.java:722)
        at org.mapdb.DBMaker.make(DBMaker.java:665)
        at edu.sdsc.scigraph.neo4j.Neo4jModule.getMaker(Neo4jModule.java:105)
        at edu.sdsc.scigraph.neo4j.Neo4jModule$$FastClassByGuice$$58219d36.invoke(<generated>)
        at com.google.inject.internal.ProviderMethod$FastClassProviderMethod.doProvision(ProviderMethod.java:230)
        at com.google.inject.internal.ProviderMethod.get(ProviderMethod.java:148)
        at com.google.inject.internal.ProviderInternalFactory.provision(ProviderInternalFactory.java:86)
        at com.google.inject.internal.InternalFactoryToInitializableAdapter.provision(InternalFactoryToInitializableAdapter.java:54)
        at com.google.inject.internal.ProviderInternalFactory.circularGet(ProviderInternalFactory.java:66)
        at com.google.inject.internal.InternalFactoryToInitializableAdapter.get(InternalFactoryToInitializableAdapter.java:46)
        at com.google.inject.internal.ProviderToInternalFactoryAdapter$1.call(ProviderToInternalFactoryAdapter.java:46)
        at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1066)
        at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
        at com.google.inject.Scopes$1$1.get(Scopes.java:65)
        at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:41)
        at com.google.inject.internal.SingleParameterInjector.inject(SingleParameterInjector.java:38)
        at com.google.inject.internal.SingleParameterInjector.getAll(SingleParameterInjector.java:62)
        at com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:107)
        at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:88)
        at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:269)
        at com.google.inject.internal.FactoryProxy.get(FactoryProxy.java:56)
        at com.google.inject.internal.ProviderToInternalFactoryAdapter$1.call(ProviderToInternalFactoryAdapter.java:46)
        at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1066)
        at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
        at com.google.inject.Scopes$1$1.get(Scopes.java:65)
        at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:41)
        at com.google.inject.internal.SingleParameterInjector.inject(SingleParameterInjector.java:38)
        at com.google.inject.internal.SingleParameterInjector.getAll(SingleParameterInjector.java:62)
        at com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:107)
        at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:88)
        at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:269)
        at com.google.inject.internal.FactoryProxy.get(FactoryProxy.java:56)
        at com.google.inject.internal.ProviderToInternalFactoryAdapter$1.call(ProviderToInternalFactoryAdapter.java:46)
        at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1066)
        at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
        at com.google.inject.Scopes$1$1.get(Scopes.java:65)
        at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:41)
        at com.google.inject.internal.SingleFieldInjector.inject(SingleFieldInjector.java:54)
        at com.google.inject.internal.MembersInjectorImpl.injectMembers(MembersInjectorImpl.java:132)
        at com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:117)
        at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:88)
        at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:269)
        at com.google.inject.internal.InjectorImpl$2$1.call(InjectorImpl.java:1009)
        at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1059)
        at com.google.inject.internal.InjectorImpl$2.get(InjectorImpl.java:1005)
        at com.google.inject.internal.InjectorImpl.getInstance(InjectorImpl.java:1044)
        at edu.sdsc.scigraph.owlapi.loader.BatchOwlLoader.load(BatchOwlLoader.java:160)
        at edu.sdsc.scigraph.owlapi.loader.BatchOwlLoader.main(BatchOwlLoader.java:192)
        ... 6 more
Caused by: java.io.IOException: Invalid argument
        at sun.nio.ch.FileChannelImpl.map0(Native Method)
        at sun.nio.ch.FileChannelImpl.map(FileChannelImpl.java:874)
        at org.mapdb.Volume$MappedFileVol.makeNewBuffer(Volume.java:493)
        ... 59 more

**Comment by ccondit** [comment]
I added a Vagrant box to the project and updated the [getting started documentation](https://github.com/SciCrunch/SciGraph#getting-started). Let me know if that solves the problem...

Note that you do not need a Neo4j server running. SciGraph currently uses an embedded Neo4j database.

Would also love to check out your Vagrant box to see what the problem might have been. Would you email me the configuration?

**Comment by bwalsh** [comment]
Thanks.  I'll email you the vagrant file I've been using.

**Comment by bwalsh** [comment]
Scratched my head for a bit, then on a hunch ( figuring that the error was happening low on the stack, close to the file system), I cloned the repo into ~/.  Previously, I had been cloning into /vagrant so that I could edit from the host system (windows).  

That was the trick.  Once I used the native file system instead of the shared volume, everything worked.

I figure it has something to do with VirtualBox's handling of the file system.  I have run into this before with node apps (VirtualBox didn't like symlinks on shared volumes).  

What this specific problem is, I don't know, but I poked around a bit and found this
 http://stackoverflow.com/questions/18370043/neo4j-server-in-virtualbox-unable-to-create-rrd-store 
 https://www.virtualbox.org/ticket/819

It looks like memory mapped files are not supported for shared folders.
